types/router_types.go

```diff
@@ -0,0 +1,28 @@
+package types
+
+// RouterType defines which type of traffic a component is handling.
+type RouterType int
+
+const (
+	RouterTypeIncoming RouterType = iota
+	RouterTypePeer
+)
+
+func (rt RouterType) String() string {
+	switch rt {
+	case RouterTypeIncoming:
+		return "incoming"
+	case RouterTypePeer:
+		return "peer"
+	default:
```

Comment:
This is going to be annoyingly nitpicky, so to make it clear I'm fine if you don't change anything.

You've set up a situation here where there's a possibility that the value is out of the defined range (although you'd have to explicitly assert an integer to get it). I was originally thinking that if it was an int, you might use it as an index into a slice of strings.

But these `Is...` functions treat it like a bool, except that out-of-range values are treated differently in the two cases.

I'd love to see one of:
* Only have either IsPeer or IsIncoming
* Define one of them using == and the other using !=
* Define one of them in terms of the other
* Use a bool instead of an int as the base type

Reply:
Thank you for pointing it out! I didn't realize the inconsistency. I changed the RouterType to be a boolean. This is much simpler. I like it